apm-agent-core/src/test/java/co/elastic/apm/agent/impl/ElasticApmTracerTest.java

```diff
@@ -119,7 +119,41 @@ void testNestedSpan() {
 
     @Test
```

Comment:
Create a class `StacktraceConfigurationTest`. Add test cases for all permutations of  `span_frames_min_duration` and `span_stack_trace_min_duration` being set/unset. Use `co.elastic.apm.agent.configuration.SpyConfiguration#createSpyConfig(org.stagemonitor.configuration.source.ConfigurationSource)` and `SimpleSource.forTest`. To create the different scenarios.

Reply:
added test